src/settings/lib/RuleEditor/rules-show-hint.js

```diff
@@ -313,50 +329,113 @@
     }
 
     cm.on('scroll', this.onScroll = function () {
-      const curScroll = cm.getScrollInfo(); const
-        editor = cm.getWrapperElement().getBoundingClientRect();
+      const curScroll = cm.getScrollInfo();
+      const editor = cm.getWrapperElement().getBoundingClientRect();
       const newTop = top + startScroll.top - curScroll.top;
       let point = newTop - (window.pageYOffset || (document.documentElement || document.body).scrollTop);
-      if (!below) point += hints.offsetHeight;
+
+      if (!below) point += container.offsetHeight;
+
       if (point <= editor.top || point >= editor.bottom) return completion.close();
-      hints.style.top = newTop + 'px';
-      hints.style.left = (left + startScroll.left - curScroll.left) + 'px';
+
+      container.style.top = newTop + 'px';
+      container.style.left = (left + startScroll.left - curScroll.left) + 'px';
     });
 
     CodeMirror.on(hints, 'dblclick', (e) => {
       const t = getHintElement(hints, e.target || e.srcElement);
-      if (t && t.hintId != null) { widget.changeActive(t.hintId); widget.pick(); }
+
+      if (t && t.hintId != null) {
+        widget.changeActive(t.hintId);
+        widget.pick();
+      }
     });
 
     CodeMirror.on(hints, 'click', (e) => {
       const t = getHintElement(hints, e.target || e.srcElement);
 
       if (t && t.hintId != null) {
         widget.changeActive(t.hintId);
-        if (completion.options.completeOnSingleClick) widget.pick();
+
+        if (completion.options.completeOnSingleClick) {
+          widget.pick();
+        }
       }
     });
 
-    CodeMirror.on(hints, 'mousedown', () => {
+    CodeMirror.on(container, 'mousedown', () => {
       setTimeout(() => { cm.focus(); }, 20);
     });
 
     CodeMirror.signal(data, 'select', completions[0], hints.firstChild);
+
     return true;
   }
 
   Widget.prototype = {
+    getHeader(className, text) {
+      const header = document.createElement("div");
+
+      if (className) {
+        header.className = className;
+      }
+      header.innerHTML = text;
+
+      return header;
+    },
+
+    getHintsList() {
+      const hints = document.createElement('ul');
+      const listItems = this.data.listDescription.list;
+
+      for (let i = 0; i < listItems.length; ++i) {
```

Comment:
Let's use `forEach`. It is more declarative.

Reply:
it looks like it shouldn't be replaced by forEach in this case because the variable **i** is used inside the code block to setup elements classes and setup hints ids. 

@VictorSoroka1 Are you sure that we should use forEach and count an additional variable manually?

Reply:
`forEach` has support for this. Plese see the example below.
`listItems.forEach((item, i) => { ... })`